sci-electronics/kicad/kicad-4.0.0_rc1.ebuild

```diff
@@ -0,0 +1,174 @@
+RDEPEND="${CDEPEND}
+	sci-electronics/electronics-menu"
+
+S="${WORKDIR}/${PN}-${MY_PV}"
+
+src_prepare() {
+	if use python; then
+		# dev-python/wxpython doesn't support python3
+		sed '/set(_PYTHON3_VERSIONS 3.3 3.2 3.1 3.0)/d' -i CMakeModules/FindPythonLibs.cmake || die "sed failed"
+	fi
+
+	# Handle optional minimal install. If minimal turned off install parts lib
+	if use minimal; then
+		sed -e '/add_subdirectory( template )/d' -i CMakeLists.txt || die "sed failed"
+	else
+		ln -s "${WORKDIR}/${P}-library" "${S}/${PN}-library" || die "ln failed"
+		sed "/add_subdirectory( bitmaps_png )/a add_subdirectory( ${PN}-library )" -i CMakeLists.txt || die "sed failed"
+		sed '/make uninstall/,/# /d' -i ${PN}-library/CMakeLists.txt || die "sed failed"
+	fi
+
+	# Add internationalization for the GUI
+	if use i18n; then
+		ln -s "${WORKDIR}/${P}-i18n" "${S}/${PN}-i18n" || die "ln failed"
+		# remove unused languages
+		local lang=""
+		for lang in ${LANGS}; do
+			if ! use linguas_${lang}; then
+				sed "/${lang}/d" -i ${PN}-i18n/LINGUAS || die "sed failed"
+			fi
+		done
+		sed "s|\${CMAKE_BINARY_DIR}|${WORKDIR}/${P}_build|g" -i ${PN}-i18n/CMakeLists.txt || die "sed failed"
+		sed "s|\${CMAKE_SOURCE_DIR}/\${LANG}|\${CMAKE_SOURCE_DIR}/${PN}-i18n/\${LANG}|g" -i ${PN}-i18n/CMakeLists.txt || die "sed failed"
+		sed "/add_subdirectory( bitmaps_png )/a add_subdirectory( ${PN}-i18n )" -i CMakeLists.txt || die "sed failed"
+		sed '/make uninstall/,$d' -i ${PN}-i18n/CMakeLists.txt || die "sed failed"
+	fi
+
+	# Install examples in the right place if requested
+	if use examples; then
+		sed -e 's:${KICAD_DATA}/demos:${KICAD_DOCS}/examples:' -i CMakeLists.txt || die "sed failed"
+	else
+		sed -e '/add_subdirectory( demos )/d' -i CMakeLists.txt || die "sed failed"
+	fi
+
+	# Add important missing doc files
+	sed -e 's/INSTALL.txt/AUTHORS.txt CHANGELOG.txt README.txt TODO.txt/' -i CMakeLists.txt || die "sed failed"
+}
+
+src_configure() {
+	need-wxwidgets unicode
+
+	local mycmakeargs=(
+		-DKICAD_DOCS="/usr/share/doc/${PF}"
+		-DKICAD_HELP="/usr/share/doc/${PF}/help"
+		-DwxUSE_UNICODE=ON
+		-DKICAD_SKIP_BOOST=ON
+		$(cmake-utils_use github BUILD_GITHUB_PLUGIN)
+		$(cmake-utils_use python KICAD_SCRIPTING)
+		$(cmake-utils_use python KICAD_SCRIPTING_MODULES)
+		$(cmake-utils_use python KICAD_SCRIPTING_WXPYTHON)
+		$(cmake-utils_use webkit KICAD_USE_WEBKIT)
+		$(usex i18n "-DKICAD_I18N_UNIX_STRICT_PATH=1" "")
+	)
+	if use debug; then
+		mycmakeargs+=(
+			-DDEBUG
+		)
+	fi
+	cmake-utils_src_configure
+}
+
+src_compile() {
+	cmake-utils_src_compile
+	if use doc; then
+		doxygen Doxyfile || die "doxygen failed"
+	fi
+}
+
+src_install() {
+	cmake-utils_src_install
+	use python && python_optimize "${D}/usr/$(get_libdir)/${EPYTHON}/site-packages/pcbnew.py"
+	if use doc ; then
+		insinto /usr/share/doc/${PF}
+		doins uncrustify.cfg
+		cd Documentation || die "cd failed"
+		doins -r GUI_Translation_HOWTO.pdf guidelines/UIpolicies.txt doxygen/*
+	fi
+}
+
+pkg_setup() {
```

Comment:
yeah, this is correct, but pkg_setup is carried out before src_prepare, so we usually move the phase there to stay consistent with the phase order (readability)